mlir/lib/Quantum/Transforms/IonsDecompositionPatterns.cpp

```diff
@@ -140,7 +140,7 @@ void cnotDecomp(catalyst::quantum::CustomOp op, mlir::PatternRewriter &rewriter)
 }
 
 std::map<std::string, std::function<void(catalyst::quantum::CustomOp, mlir::PatternRewriter &)>>
-    funcMap = {{"T", &tDecomp},        {"S", &sDecomp},   {"Z", &zDecomp},
+    funcMap = {{"T", &tDecomp},        {"S", &sDecomp},   {"PauliZ", &zDecomp},
```

Comment:
```suggestion
    funcMap = {{"T", &tDecomp},        {"S", &sDecomp},   {"Z", &zDecomp}, {"PauliZ", &zDecomp},
```

I think we should have both as `qml.Z` and `qml.PauliZ` are both valid.

Reply:
Discussed, this is not necessary since `qml.Z` also gets lowered to `quantum.custom "PauliZ"`